### DISCUSSION TITLE
Reaching fields out of a listener of a Repeater

### DISCUSSION THREAD

**Discussion by antoinemy** [filamentphp/filament#5309] [Help] [answered]
Hi all,

I need to update the sum of the data present in my Repeater when I add, delete or clone a row in it. Let's take the example of invoices, when I delete a product, the total sum is updated by deducting the price of the deleted product... Etc...

I dug here:

```
Repeater::make('table')
  ...
  ->registerListeners([
      'repeater::createItem' => [
          function (Repeater $component, string $statePath): void {
              
          },
      ],
      ...
  ])
```

And I would have liked something as simple as the afterStateUpdated() function with the Closure $get, $set...:

```
TextInput::make('...')
  ->reactive()
  ->afterStateUpdated(function (Closure $get, $set, $state) {
      $get('input1'); 
      $set('input2', '...');
  })
```
But I can't reach and edit the other elements... any ideas?

Thank's

**Comment by danharrin** [comment, marked as answer]
You can probably get `$get, $set, $state` within the listener closure? Have you tried?

**Reply by antoinemy** [reply]
I tried : 

```
...
'repeater::createItem' => [
    function (Closure $get, $set, $state): void {
       dd($get);
    },
],
...
```

and it doesn't work, any ideas?

`
App\Filament\Resources\RecipeResource::App\Filament\Resources\{closure}(): Argument #1 ($get) must be of type Closure, Filament\Forms\Components\Repeater given, called in /var/www/html/vendor/filament/forms/src/Components/Concerns/ListensToEvents.php on line 12
`

**Reply by danharrin** [reply]
Oh weird, ok. Scrap my idea and use $component.

You can def access `$get / $set / $state` within other closures if you use something like `$component->evaluate(fn ($get) => $get('...'))`

**Reply by antoinemy** [reply]
Thank you very much.

For those who are stuck :

```
...
'repeater::createItem' => [
  function (Repeater $component, string $statePath): void {
      $component->evaluate(function ($get, $set, $state) {
          // Do stuff...
      });
  },
],
...
```
